server/src/main/java/org/opensearch/cluster/health/ClusterStateHealth.java

```diff
@@ -147,6 +149,7 @@ public ClusterStateHealth(final StreamInput in) throws IOException {
         unassignedShards = in.readVInt();
         numberOfNodes = in.readVInt();
         numberOfDataNodes = in.readVInt();
+        hasDiscoveredMaster = in.readBoolean();
```

Comment:
I believe we should have Version Check here on input stream to allow backward compatibility and prevent serialisation issues.

Reply:
Done.